docs/005-mobile-app-architecture.adoc

```diff
@@ -0,0 +1,153 @@
+** The route *identifies* a screen. Each screen should have a route, i.e. if we name a file `[...]_screeen` we expect it to be routable.
+** There is no clear definition that distinguishes pages from screens. I would recommend not to use page because it is yet another term that adds confusion.
+* Service: A service is on the application level. We should treat the data processing through the native rust backend as service. If we e.g. periodically fetch data we should create a service in Dart that handles that. Note that some of the Dart code related to the service will be generated by Flutter Rust bridge, and code that we need on top (to e.g. periodically fetch information from the native rust code) should be added as service.
+
+Recommended initial folder structure:
+
+* lib
+** common
+*** application
+**** `some_shared_service.dart`
+*** domain
+**** `some_shared_domain_class.dart`
+*** `some_presentation_layer_class.dart`
+** features
+*** wallet
+**** application
+***** `some_shared_service.dart`
+**** domain
+***** `some_shared_domain_class.dart`
+**** `some_presentation_layer_class.dart`
+*** trade
+**** application
+***** `some_shared_service.dart`
+**** domain
+***** `some_shared_domain_class.dart`
+**** `some_presentation_layer_class.dart`
+
+This image shows the boundaries of the Flutter layers:
+
+.Flutter Layer Overview (https://codewithandrea.com/articles/flutter-project-structure/[source])
+image::assets/006-mobile-application-architecture/flutter-layers.png[alt=FlutterLayers,width=400,height=480]]
+
+Within the `wallet` and `trade` features we can introduce subfolders:
+
+* `application` - services; we will likely not need this in Dart because we handle it in rust
+* `domain` - domain model files
+
+Additionally, we could add:
+
+* `presentation` - widget related files (including widget state and widget controllers); this might be unnecessary, we could also just put these files in the feature folder directly
+* `data` - storage related files; we will likely not need this in Dart
+
+Initially it is expected that almost all of our files except from commonly share widgets / models will be in the `presentation` folder, so I would not introduce this folder structure, but rather add all files to the `wallet` folder.
+
+We can opt to further structure a features subfolder into sub-feature folders (e.g. `send` and `receive` for the `wallet` folder), but I would do that once the application grows.
+
+It is recommended to name widgets semantically by context (feature, task) and a logical suffix that is associated with the composition of the widget.
+Examples:
+
+- `[...]_sceen`: composition of widgets that represents a screen (identified by a route)
+    - Examples: `wallet_screen`, `trade_screen`, `scanner_screen`
+    - There is not clear guideline for screens that are part of a wizard yet, I would recommend numbering hem (e.g. `send_screen_1`)
```

Comment:
How about creating a sub folder e.g. funding wizard. All screens within that folder belong to that wizard.